src/test/shared/sam/cli/samCliVersion.test.ts

```diff
@@ -0,0 +1,75 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import * as semver from 'semver'
+import { SamCliVersion, SamCliVersionValidation } from '../../../../shared/sam/cli/samCliVersion'
+
+describe('SamCliVersion', async () => {
+
+    it('validates undefined version', async () => {
+        assert.equal(SamCliVersion.validate(), SamCliVersionValidation.VersionNotParseable)
+    })
+
+    it('validates valid versions', async () => {
+        [
+            SamCliVersion.MINIMUM_SAM_CLI_VERSION_INCLUSIVE,
+            '0.7.1',
+            '0.8.0',
+            '0.10.999',
+        ].forEach(version => {
+            const validation = SamCliVersion.validate(version)
+            assert.equal(
+                validation, SamCliVersionValidation.Valid,
+                `Version: ${version}, Validation: ${SamCliVersionValidation[validation]}`
```

Comment:
Minor: As a developer, it becomes instinct to avoid repeating yourself as much as possible. But when writing tests, this can come back to bite you. If there's a bug in `SamCliVersionValidation`, these tests will fail to detect it. Instead, suggest to explicitly list the expected result (true or false) for each version in the test case. The extra safety is worth the extra brittleness.